pkg/devserver/api.go

```diff
@@ -202,11 +202,18 @@ func (a devapi) register(ctx context.Context, r sdk.RegisterRequest) (*cqrs.Sync
 		return nil, publicerr.Wrap(err, 400, "Invalid request")
 	}
 
+	// TODO Retrieve same syncID for connect, if r.IdempotencyKey is the same
+	syncID := uuid.New()
+
 	if app, err := a.devserver.Data.GetAppByChecksum(ctx, consts.DevServerEnvID, sum); err == nil {
 		if !app.Error.Valid {
 			// Skip registration since the app was already successfully
 			// registered.
-			return &cqrs.SyncReply{OK: true}, nil
+			return &cqrs.SyncReply{
+				OK:     true,
+				AppID:  &app.ID,
```

Comment:
We always, always, always have to return the App ID in case of a connect sync. Otherwise, we can't associate connections with apps and routing will fail